src/loop/parser.ml

```diff
@@ -190,70 +192,131 @@ module Make(State : State.S) = struct
       let st = State.error st ~file ~loc:{ file = file.loc; loc; } parsing_error (syntax_error_ref, perr) in
       st, None
 
-  let parse_logic prelude st (file : Logic.language file) =
+  let[@inline] infer_lang source (file : Logic.language file) =
+    match source with
+    | `Raw (filename, _) | `File filename ->
+      (* Auto-detect input format *)
```

Comment:
maybe move that comment inside the `None` case, where we actually do the auto-detection ?